src/functions.cpp

```diff
@@ -29,20 +29,22 @@
 #include "wincrypt.h"
 #endif
 
-#define ARG(argname, argtype) get_arg<argtype>(argname, env, sig, pstate, backtrace)
-#define ARGR(argname, argtype, lo, hi) get_arg_r(argname, env, sig, pstate, lo, hi, backtrace)
-#define ARGM(argname, argtype, ctx) get_arg_m(argname, env, sig, pstate, backtrace, ctx)
+#define ARG(argname, argtype) get_arg<argtype>(argname, env, sig, params, pstate, backtrace)
+#define ARGR(argname, argtype, lo, hi) get_arg_r(argname, env, sig, params, pstate, lo, hi, backtrace)
+#define ARGM(argname, argtype, ctx) get_arg_m(argname, env, sig, params, pstate, backtrace, ctx)
 
 namespace Sass {
   using std::stringstream;
   using std::endl;
 
-  Definition* make_native_function(Signature sig, Native_Function func, Context& ctx)
+  Definition* make_native_function(std::string name, Signature sig, Parameters* params, Native_Function func, Context& ctx)
   {
-    Parser sig_parser = Parser::from_c_str(sig, ctx, ParserState("[built-in function]"));
-    sig_parser.lex<Prelexer::identifier>();
-    std::string name(Util::normalize_underscores(sig_parser.lexed));
-    Parameters* params = sig_parser.parse_parameters();
+    if (params == 0) {
```

Comment:
Skip the prototype parsing if `Parameters` are passed to us to gain performance.